### ISSUE TITLE
Importing types doesn't trigger an error in checkJs

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#22159]
```ts
// ./a.ts
export type Hello = string;
```

```js
import { Hello } from "./a";
```

**Expected:** Error: Types cannot be imported in a JavaScript file.
**Actual:** No error.

**Comment by mhegazy** [comment]
PRs welcomed.
